src/Model/Consignment/PostNLConsignment.php

```diff
@@ -296,6 +296,31 @@ public function setAgeCheck(bool $ageCheck): AbstractConsignment
         return $this;
     }
 
+    /**
+     * @return bool
+     */
+    public function hasCooledDelivery(): bool
+    {
+        return (bool) $this->cooled_delivery;
+    }
+
+    /**
+     * Cooled Delivery
+     *
+     * Required: No
+     *
+     * @param bool $cooledDelivery
+     *
+     * @return $this
+     * @throws \Exception
+     */
+    public function setCooledDelivery(bool $cooledDelivery): AbstractConsignment
```

Comment:
Waarom is de return type de parent class en niet deze?

Reply:
Omdat de declaratie van de return type compatibel moet zijn met AbstractConsignment->setCooledDelivery(cooledDelivery: bool) : \MyParcelNL\Sdk\src\Model\Consignment\AbstractConsignment. 

Vanaf PHP 7.4 zal dat wel mogelijk zijn.

Reply:
Je kan toch ook allebei return self gebruiken?

Reply:
Helaas werkt dat niet